dts/bindings/regulator/regulator-gpio.yaml

```diff
@@ -0,0 +1,43 @@
+# Copyright 2023 EPAM Systems
+# SPDX-License-Identifier: Apache-2.0
+
+description: GPIO-controlled voltage of regulators
+
+include:
+  - name: base.yaml
+  - name: regulator.yaml
+    property-allowlist:
+      - regulator-name
+      - regulator-min-microvolt
+      - regulator-max-microvolt
+
+compatible: "regulator-gpio"
+
+properties:
+  regulator-name:
+    required: true
+
+  regulator-min-microvolt:
+    required: true
+
+  regulator-max-microvolt:
+    required: true
+
+  switch-voltage-gpios:
```

Comment:
please keep compatibility with https://www.kernel.org/doc/Documentation/devicetree/bindings/regulator/gpio-regulator.txt, ie use `gpios` and add any missing property, e.g. `startup-delay-us` (except `enable-active-high`, seems to be obsolete).

Reply:
Will not copy-pasting everything from Linux kernel be breaking of license rules?